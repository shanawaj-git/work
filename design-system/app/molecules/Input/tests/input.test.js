import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { unmountComponentAtNode } from 'react-dom';
import { render } from 'react-testing-library';

import { Input } from '../index';

const shallowRenderer = new ShallowRenderer();
let container = null;

describe('Input.js', () => {
  const props = {
    name: 'testInput',
    variant: 'standard',
    onChange: jest.fn(),
  };

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  describe('index', () => {
    it('Should import succesfully', () => {
      const tree = render(<Input {...props} />);
      expect(tree).toBeTruthy();
      expect(tree.container).toMatchSnapshot();
    });

    it('Should display a disabled input field', () => {
      shallowRenderer.render(<Input {...props} disabled />);
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });

    it('Should display an input field for upload files', () => {
      const tree = render(<Input {...props} type="file" />);
      expect(tree).toBeTruthy();
      expect(tree.container).toMatchSnapshot();
    });

    it('Should have onBlur and onFocus if passed', () => {
      const dummyFun = jest.fn();
      const input = {
        name: 'test',
        onFocus: dummyFun,
        onBlur: dummyFun,
        onChange: jest.fn(),
      };
      shallowRenderer.render(<Input {...props} input={input} />);
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });

    it('Should have a label if passed', () => {
      shallowRenderer.render(<Input {...props} label="Label" />);
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });

    it('Should show an input with a prefix', () => {
      shallowRenderer.render(<Input {...props} prefix="USD" />);
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });
  });
});
