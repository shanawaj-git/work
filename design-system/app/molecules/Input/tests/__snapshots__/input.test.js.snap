// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.js index Should display a disabled input field 1`] = `
<FormControlWrapper
  className=""
  disabled={true}
  error={false}
  errorText=""
  helperText=""
>
  <ForwardRef(Styled(Component))
    InputLabelProps={
      Object {
        "classes": Object {},
        "shrink": true,
      }
    }
    InputProps={
      Object {
        "classes": Object {
          "focused": undefined,
          "input": undefined,
        },
        "endAdornment": null,
        "startAdornment": null,
      }
    }
    autoComplete=""
    autoFocus={false}
    disabled={true}
    error={false}
    inputProps={
      Object {
        "name": "testInput",
      }
    }
    inputRef={null}
    multiline={false}
    name="testInput"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    placeholder=""
    type="text"
    variant="standard"
  />
</FormControlWrapper>
`;

exports[`Input.js index Should display an input field for upload files 1`] = `
<div>
  <div
    class="root MuiFormControl-root styledComponents__FormControl-lwe1h6-0 jHalSF css-1nrlq1o-MuiFormControl-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1vly5ry-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete=""
          class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
          name="testInput"
          placeholder=""
          type="file"
          value=""
        />
        <div
          class="root MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium styledComponents__InputAdornment-sc-1swdb7r-0 gMjvtV css-1laqsz7-MuiInputAdornment-root"
        >
          <img
            src="IMAGE_MOCK"
          />
        </div>
      </div>
    </div>
    
  </div>
</div>
`;

exports[`Input.js index Should have a label if passed 1`] = `
<FormControlWrapper
  className=""
  disabled={false}
  error={false}
  errorText=""
  helperText=""
>
  <ForwardRef(Styled(Component))
    InputLabelProps={
      Object {
        "classes": Object {},
        "shrink": true,
      }
    }
    InputProps={
      Object {
        "classes": Object {
          "focused": undefined,
          "input": undefined,
        },
        "endAdornment": null,
        "startAdornment": null,
      }
    }
    autoComplete=""
    autoFocus={false}
    disabled={false}
    error={false}
    inputProps={
      Object {
        "name": "testInput",
      }
    }
    inputRef={null}
    label={
      <React.Fragment>
        Label
      </React.Fragment>
    }
    multiline={false}
    name="testInput"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    placeholder=""
    type="text"
    variant="standard"
  />
</FormControlWrapper>
`;

exports[`Input.js index Should have onBlur and onFocus if passed 1`] = `
<FormControlWrapper
  className=""
  disabled={false}
  error={false}
  errorText=""
  helperText=""
>
  <ForwardRef(Styled(Component))
    InputLabelProps={
      Object {
        "classes": Object {},
        "shrink": true,
      }
    }
    InputProps={
      Object {
        "classes": Object {
          "focused": undefined,
          "input": undefined,
        },
        "endAdornment": null,
        "startAdornment": null,
      }
    }
    autoComplete=""
    autoFocus={false}
    disabled={false}
    error={false}
    inputProps={
      Object {
        "name": "test",
      }
    }
    inputRef={null}
    multiline={false}
    name="testInput"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder=""
    type="text"
    variant="standard"
  />
</FormControlWrapper>
`;

exports[`Input.js index Should import succesfully 1`] = `
<div>
  <div
    class="root MuiFormControl-root styledComponents__FormControl-lwe1h6-0 jHalSF css-1nrlq1o-MuiFormControl-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1vly5ry-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete=""
          class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          name="testInput"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
    
  </div>
</div>
`;

exports[`Input.js index Should show an input with a prefix 1`] = `
<FormControlWrapper
  className=""
  disabled={false}
  error={false}
  errorText=""
  helperText=""
>
  <ForwardRef(Styled(Component))
    InputLabelProps={
      Object {
        "classes": Object {},
        "shrink": true,
      }
    }
    InputProps={
      Object {
        "classes": Object {
          "focused": undefined,
          "input": undefined,
        },
        "endAdornment": null,
        "startAdornment": <ForwardRef(styledComponents__InputAdornment)
          classes={
            Object {
              "root": undefined,
            }
          }
          position="start"
        >
          USD
        </ForwardRef(styledComponents__InputAdornment)>,
      }
    }
    autoComplete=""
    autoFocus={false}
    disabled={false}
    error={false}
    inputProps={
      Object {
        "name": "testInput",
      }
    }
    inputRef={null}
    multiline={false}
    name="testInput"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    placeholder=""
    type="text"
    variant="standard"
  />
</FormControlWrapper>
`;
